receiver/stream_bms_receiver_test.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+from cal_para_values import min_max_smavg
+import read_input
+import cal_para_values
+
+class test_bms_stream_receiver(unittest.TestCase):
+
+    def test_to_check_format(self):
+
+        self.assertTrue(read_input.check_data_type({"temperature":0.0,"soc":62.903126,"charge_rate":0.045999337}) == 'data_is_correct' )
+
+    def test_to_check_wrong_format(self):   
+        
+        self.assertTrue(read_input.check_data_type({"temperature":0.0,"soc":'abcd',"charge_rate":0.045999337}) == 'data_is_not_correct' )
+
+        self.assertTrue(read_input.check_data_type({"temperature":0.0,"soc":62.903126,"charge_rate":'abcd'}) == 'data_is_not_correct' )
+
+        self.assertTrue(read_input.check_data_type({"temperature":'abcd',"soc":62.903126,"charge_rate":0.9}) == 'data_is_not_correct' )
+
+    def test_to_check_cal_values(self):
+
+        self.assertTrue(cal_para_values.cal_values([0,1,2,3,6]) == { 'min' :0, 'max': 6, 'simple moving average':2.4})
+        self.assertTrue(cal_para_values.cal_values([0,1,2,3]) == { 'min' :0, 'max': 3})
+
+    def test_to_check_false_cal_values(self):
+        self.assertFalse(cal_para_values.cal_values([0,1,2,3]) == { 'min' :0, 'max': 3, 'simple moving average':1.2})
+
+        self.assertFalse(cal_para_values.cal_values([0,2,4,8]) == { 'min' :0, 'max': 4})
+
+        self.assertFalse(cal_para_values.cal_values([0,1,5,3]) == { 'min' :1, 'max': 5})
+
+        self.assertFalse(cal_para_values.cal_values([2,0.01,0.8232,0.2583,3]) == { 'min' :0.01, 'max': 3, 'simple moving average':1.2163} )

```

Comment:
This test is ok for comparing equality. But if the average comes out like (5+3+2)/3, it will be 3.3333....
Better to use a tolerance epsilon when comparing floating-point values.